modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/PagedReflectiveOperationExecutorFactory.java

```diff
@@ -49,18 +49,23 @@ public PagedReflectiveOperationExecutorFactory(Class<T> implementationClass, Met
                                                    createDelegate(implementationClass, parameters), pagingProviderTransformer);
   }
 
-  private class PagingProviderResultTransformer implements BiFunction<Object, ExecutionContext<M>, Object> {
+  private class PagingProviderResultTransformer implements CheckedBiFunction<Object, ExecutionContext<M>, Object> {
 
     @Inject
     private ExtensionConnectionSupplier connectionSupplier;
 
     @Override
-    public Object apply(Object result, ExecutionContext<M> executionContext) {
+    public Object applyChecked(Object result, ExecutionContext<M> executionContext) throws Throwable {
       PagingProviderProducer pagingProviderProducer =
           new PagingProviderProducer<>((PagingProvider) result, executionContext.getConfiguration().get(),
                                        (ExecutionContextAdapter) executionContext, connectionSupplier);
       ListConsumer<?> consumer = new ListConsumer(pagingProviderProducer);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use try-with-resources or close this "ListConsumer" in a "finally" clause. ([squid:S2095](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2095))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7717&issues=AWkazdnfotIXnmsUFhSz&open=AWkazdnfotIXnmsUFhSz)

[ISSUE_KEY]: AWkazdnfotIXnmsUFhSz

Reply:
this is not valid for this use case, we are retuning the consumer inside the ConsumerStreamingIterator